spec/03-plugins/37-opentelemetry/04-exporter_spec.lua

```diff
@@ -0,0 +1,166 @@
+require "kong.plugins.opentelemetry.proto"
+local helpers = require "spec.helpers"
+local utils = require "kong.tools.utils"
+local tablex = require "pl.tablex"
+local pb = require "pb"
+
+local table_merge = utils.table_merge
+local HTTP_PORT = 35000
+
+for _, strategy in helpers.each_strategy() do
+  describe("opentelemetry exporter #" .. strategy, function()
+    local proxy_client
+
+    lazy_setup(function ()
+      -- overwrite for testing
+      pb.option("enum_as_value")
+    end)
+
+    lazy_teardown(function()
+      -- revert it back
+      pb.option("enum_as_name")
+    end)
+
+    -- helpers
+    local function setup_instrumentations(types, config)
+      local bp, _ = assert(helpers.get_db_utils(strategy, {
+        "services",
+        "routes",
+        "plugins",
+      }, { "opentelemetry" }))
+
+      local http_srv = assert(bp.services:insert {
+        name = "mock-service",
+        host = helpers.mock_upstream_host,
+        port = helpers.mock_upstream_port,
+      })
+
+      bp.routes:insert({ service = http_srv,
+                         protocols = { "http" },
+                         paths = { "/" }})
+
+      bp.plugins:insert({
+        name = "opentelemetry",
+        config = table_merge({
+          endpoint = "http://127.0.0.1:" .. HTTP_PORT,
+          batch_flush_delay = -1, -- report immediately
+        }, config)
+      })
+
+      assert(helpers.start_kong {
+        database = strategy,
+        nginx_conf = "spec/fixtures/custom_nginx.template",
+        plugins = "opentelemetry",
+        opentelemetry_tracing = types,
+      })
+
+      proxy_client = helpers.proxy_client(1000)
+    end
+
+    describe("valid #http request", function ()
+      lazy_setup(function()
+        os.execute("fuser -n tcp -k " .. HTTP_PORT)
```

Comment:
little worried about the leftover behaviour here, is it needed? if so maybe move to lazy_teardown 🤔

Reply:
Moved to lazy_teardown.